src/components/HeatmapContainer.js

```diff
@@ -1,31 +1,114 @@
+    case 34: colsCSS = "grid-cols-34"; break; 
+    case 35: colsCSS = "grid-cols-35"; break; 
+    case 36: colsCSS = "grid-cols-36"; break; 
+    case 37: colsCSS = "grid-cols-37"; break; 
+    case 38: colsCSS = "grid-cols-38"; break; 
+    case 39: colsCSS = "grid-cols-39"; break; 
+    case 40: colsCSS = "grid-cols-40"; break; 
+  }
+  return colsCSS;
+}
+
+const HeatmapContainer = ({
+  data,
+  highlightProperties,
+  isPropertiesHighlighted,
+  height,
+  width,
+}) => {
   const cells = [];
-  const generateCells = () => {
-    Array(data.length).fill(1)
-      .map((ie, i) => cells.push({info: data[i], cell: <HeatmapCell value={data[i].value} state={heatmapCellStates.default}/>})
+  Array(data.length)
+    .fill(1)
+    .map((ie, i) =>
+      cells.push({
+        info: data[i],
+        cell: (
+          <HeatmapCell
+            value={data[i].value}
+            state={heatmapCellStates.default}
+            info={data[i]}
+          />
+        ),
+      })
     );
-  }
 
-  generateCells();
-  if (isPropertiesHighlighted) 
+  if (isPropertiesHighlighted) {
     for (const highlightProperty of highlightProperties) {
-      cells.map((x) => cells.map((x, i) => {
-        if (!x.info.wins.includes(highlightProperty) && !x.info.losses.includes(highlightProperty)) x.cell =  <HeatmapCell key={Math.random()*1000000} value={cells[i].info.value} state={heatmapCellStates.isDimmed} info={x.info}/>
-        else x.cell = <HeatmapCell key={Math.random()*1000000} value={cells[i].info.value} state={heatmapCellStates.isHighlighted} info={x.info}/>
-      }))
+      cells.map((x) =>
+        cells.map((x, i) => {
+          if (
+            !x.info.wins.includes(highlightProperty) &&
+            !x.info.losses.includes(highlightProperty)
+          )
+            x.cell = (
+              <HeatmapCell
+                key={Math.random() * 1000000}
+                value={cells[i].info.value}
+                state={heatmapCellStates.isDimmed}
+                info={x.info}
+              />
+            );
+          else
+            x.cell = (
+              <HeatmapCell
+                key={Math.random() * 1000000}
+                value={cells[i].info.value}
+                state={heatmapCellStates.isHighlighted}
+                info={x.info}
+              />
+            );
+        })
+      );
     }
+  }
+
+  const numCols = Math.floor((width + HeatmapContainerGap) / (HeatmapCellSize + HeatmapContainerGap));
+  const numRows = Math.floor((height + HeatmapContainerGap) / (HeatmapCellSize + HeatmapContainerGap));
+  const cellsToRender = cells.slice(- numRows * numCols);
 
-  return(
-    <div className="grid grid-cols-24 xl:grid-cols-24 lg:grid-cols-22 2xl:grid-cols-26 gap-8 p-4 max-w-full" style={{
-            backgroundColor: `rgba(10, 10, 20)`
-        }}>
-      {cells.map(x => x.cell)}
+  return (
+    <div
+      className={`grid ${getColsCSS(numCols)} ${HeatmapContainerGapCSS} p-4 max-w-full`}
```

Comment:
use style={{}} for the variable number of columns

Reply:
style={{grid-template-columns: `repeat(${numCols}, minmax(0, 1fr)`)}}



grid-template-columns: repeat(8, minmax(0, 1fr));


<br class="Apple-interchange-newline" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none;">



grid-template-columns: repeat(8, minmax(0, 1fr));


<br class="Apple-interchange-newline" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none;">



grid-template-columns: repeat(8, minmax(0, 1fr));


<br class="Apple-interchange-newline" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none;">